rest/rest_collection_impl.h

```diff
@@ -615,6 +616,7 @@ addBackgroundJobInThread(Key key,
 
         if (impl->entries.cmp_xchg(oldEntries, newEntries, true)) {
             // Now we can start the task, since the commit succeeded
+            task->running = true;
```

Comment:
As long as it doesn't enter this condition scope, the task was not really running, hence the deadlock.